libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -89,6 +90,10 @@ internal class RouteAlternativesController constructor(
             // the call to directionsSession.setRoutes
             return emptyList()
         }
+
+        override fun onError(message: String) {
+            logE(TAG, Message("Error: $message"))
+        }
```

Comment:
Are there any cases where we should call the observer with empty routes here (to cleanup the state after error)?

Reply:
Discussing this with @LukasPaczos we have some questions @mapbox/navnative:

- are there situations where we should clean everything up?
- what type of errors can it report? just failed requests or something more?

Reply:
cc @DenisPryt

Reply:
We report message from RouterError now (i.e. we report only about RouterError) 
If this error occurs you don't need to clean up the state (At least now). If we get this error it means that we've been trying to request new routes but something went wrong.

Reply:
onRouteAlternativesChanged will be invoked after alternatives changed regardless of the errors that occur.
If we pass alternatives after error, onRouteAlternativesChanged will be invoked. Or if alternatives will changed for any reasons (refreshImmediately or timeout) -- onRouteAlternativesChanged will be invoked (only if alternatives will be changed)

Reply:
Awesome! Thanks for confirming @DenisPryt 🙇‍♂️